### DISCUSSION TITLE
Typescript ES2019

### DISCUSSION THREAD

**Discussion by Winteriver** [gulpjs/gulp#2459] [Help] [answered]
Gulp version: 
`CLI version: 2.2.0,
Local version: 4.0.2`

When using "es2019" as lib in compilerOptions it says:
`error TS6046: Argument for '--lib' option must be: 'es5', 'es6', 'es2015', 'es7', 'es2016', 'es2017', 'es2018', 'esnext', 'dom', 'dom.iterable', 'webworker', 'webworker.importscripts', 'scripthost', 'es2015.core', 'es2015.collection', 'es2015.generator', 'es2015.iterable', 'es2015.promise', 'es2015.proxy', 'es2015.reflect', 'es2015.symbol', 'es2015.symbol.wellknown', 'es2016.array.include', 'es2017.object', 'es2017.sharedmemory', 'es2017.string', 'es2017.intl', 'es2017.typedarrays', 'es2018.intl', 'es2018.promise', 'es2018.regexp', 'esnext.array', 'esnext.symbol', 'esnext.asynciterable', 'esnext.intl', 'esnext.bigint'.`

But I need it for functions like map, or flapMap, otherwise (es2018 and below) I get:
`error TS2339: Property 'flatMap' does not exist on type (...)`

So, how do I Use ECMAScript 2019 with gulp?

**Comment by phated** [comment]
🤷 We don't do anything with typescript, I assume your environment is messed up. Moving this to the discussion forum, in case a community member can help you.

**Reply by Winteriver** [reply]
Thank you :)
The script execution works and compiler in VS Studio code has no errors - its only the gulp error above.

**Comment by Winteriver** [comment, marked as answer]
Okay, Error is gone when using "esnext" instead of "es2019".

Thanks!
